datarobot_batch_scoring/batch_scoring.py

```diff
@@ -412,23 +416,53 @@ def __iter__(self):
                                'we lost {} records'.format(
                                    batch.id, len(batch.data)))
                 continue
-            hook = partial(self._response_callback, batch=batch)
 
             if self.fast_mode:
                 chunk_formatter = fast_to_csv_chunk
             else:
                 chunk_formatter = slow_to_csv_chunk
 
-            data = chunk_formatter(batch.data, batch.fieldnames)
-            self._ui.debug('batch {} transmitting {} bytes'
-                           .format(batch.id, len(data)))
-            yield requests.Request(
-                method='POST',
-                url=self.endpoint,
-                headers=self.headers,
-                data=data,
-                auth=(self.user, self.api_token),
-                hooks={'response': hook})
+            todo = [batch]
+            while todo:
+                batch = todo.pop(0)
+                data = chunk_formatter(batch.data, batch.fieldnames)
+                if len(data) < self.max_batch_size:
```

Comment:
Server will reject batches larger than 10mb, and this thing tries to prevent exactly this thing. We are not trying to format all batches to exactly X megs, we just check final size of encoded batch with 0 overhead, and do something only if it's size bigger than limit.

It's not supposed to happen often